theano/misc/pycuda_example.py

```diff
@@ -285,6 +285,11 @@ def __init__(self, scalar_op, inplace_pattern=None, name=None):
         self.scalar_op = scalar_op
         self.inplace_pattern = inplace_pattern
 
+    # As we have a dict in props, we need to implement __hash__
+    def __hash__(self):
+        return (hash(type(self)) ^ hash(self.scalar_op) ^
+                hash_from_dict(self.inplace_pattern))
```

Comment:
Is that the right way? Or should the default `__hash__` method be updated to call `hash_from_dict` instead?